sql/2024/fonts/performance/fonts_size.sql

```diff
@@ -0,0 +1,35 @@
+-- Section: Performance
+-- Question: What is the distribution of the file size?
+-- Normalization: Fonts
+
+WITH
+fonts AS (
+  SELECT
+    client,
+    url,
+    PARSE_NUMERIC(JSON_EXTRACT_SCALAR(ANY_VALUE(summary), '$.respBodySize')) AS size
+  FROM
+    `httparchive.all.requests`
+  WHERE
+    date = '2024-07-01' AND
+    is_root_page AND
+    type = 'font'
+  GROUP BY
+    client,
+    url
+)
+
+SELECT
+  client,
+  percentile,
+  COUNT(0) AS count,
+  ROUND(APPROX_QUANTILES(size, 1000)[OFFSET(percentile * 10)]) AS size
+FROM
+  fonts,
+  UNNEST([10, 25, 50, 75, 90, 99]) AS percentile
```

Comment:
@IvanUkhov We used `10, 25, 50, 75, 90, 100` in the 2022 query. I think this is better though, but we should make a note of it in the chapter.

Reply:
Yeah, the 100th percentile is kind of meaningless. It just picks up outliers, which one is not going to show in the report anyway. I am even regretting I did not go lower to, say, 95 in all queries calculating percentiles.

Reply:
We normally show 10, 25, 50, 75, 90. The 100 is for the top stat which can some times make for a funny throw away line (“Shockingly the largest amount of fonts loaded by a single page is 123,456”) but we don’t normally include them in the actual charts (in part because they often skew the access).

Reply:
Yea, we didn't use the 100th percentile in the 2022 article, but it was in the spreadsheet. I was just pointing out the difference in the query as we were internally discussing the difference between the 2022 data and 2024 data which I think can be explained by this change.

Reply:
Well, given my understanding for what this is used for in the past, I'm not sure 99 (or 95) is better than 100. But  up to you all!